### PR TITLE
Subtask/as 86 connect question to form data

### PR DESCRIPTION
**Pull request by swdpcomputing**
## Ticket Number

AS-86

## Description of change

created post, added question to form data

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by davehlewis — CHANGES_REQUESTED** [review]

File: packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js [hunk 1]
```diff
@@ -1,9 +1,67 @@
...
+  appealReply.sectionStates[sectionName][taskName] = getTaskStatus(
+    appealReply,
+    sectionName,
+    taskName
+  );
+
+  try {
+    req.session.appealReply = await createOrUpdateAppealReply(appealReply);
+  } catch (e) {
+    logger.error(e);
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js:53]: Done

File: packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js [hunk 2]
```diff
@@ -1,9 +1,67 @@
+const logger = require('../lib/logger');
 const { VIEW } = require('../lib/views');
 const getAppealSideBarDetails = require('../lib/appeal-sidebar-details');
+const { getTaskStatus } = require('../services/task.service');
+const { createOrUpdateAppealReply } = require('../lib/appeal-reply-api-wrapper');
+
+const sectionName = 'aboutAppealSection';
+const taskName = 'extraConditions';
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js:8]: This is the same as other-appeals so not sure what this refers to.

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js [hunk 3]
```diff
@@ -7,35 +11,160 @@ jest.mock('../../../src/services/task.service');
...
-    extraConditions.getExtraConditions(req, res);
+    extraConditionsController.getExtraConditions(req, res);
 
     expect(res.render).toHaveBeenCalledWith(VIEW.EXTRA_CONDITIONS, {
       appeal: null,
       backLink: `/mock-id/${VIEW.TASK_LIST}`,
     });
   });
+
+  describe('postOtherAppeals', () => {
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js:50]: Done

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js [hunk 4]
```diff
@@ -7,35 +11,160 @@ jest.mock('../../../src/services/task.service');
...
+    it('should redirect with extra-conditions set to false', async () => {
+      getTaskStatus.mockImplementation(() => mockTaskStatus);
+
+      mockAppealReply.aboutAppealSection.extraConditions.hasExtraConditions = false;
+      // mockAppealReply.sectionStates.extraConditions.extraConditions = mockTaskStatus;
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js:55]: Not needed - deleted.

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js [hunk 5]
```diff
@@ -7,35 +11,160 @@ jest.mock('../../../src/services/task.service');
...
+      expect(createOrUpdateAppealReply).toHaveBeenCalledWith(mockAppealReply);
+      expect(res.render).not.toHaveBeenCalled();
+      expect(res.redirect).toHaveBeenCalledWith(`/mock-id/${VIEW.TASK_LIST}`);
+    });
+
+    it('should redirect to the back link specified', async () => {
+      getTaskStatus.mockImplementation(() => mockTaskStatus);
+
+      mockAppealReply.aboutAppealSection.extraConditions.hasExtraConditions = false;
+      // mockAppealReply.sectionStates.aboutAppealSection.otherAppeals = mockTaskStatus;
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js:75]: Not needed - deleted.

File: packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davehlewis) [reply packages/lpa-questionnaire-web-app/src/controllers/extra-conditions.js:67]: Not sure if this was missed but still needs updating

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (davehlewis) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/extra-conditions.test.js:106]: Naming is still incorrect here

**Review by davehlewis — CHANGES_REQUESTED** [review]

**Review by davehlewis — APPROVED** [review]

**Pull request merged**